Library/Homebrew/livecheck.rb

```diff
@@ -77,24 +77,48 @@ def strategy(symbol = nil)
   # Sets the `@url` instance variable to the provided argument or returns the
   # `@url` instance variable when no argument is provided. The argument can be
   # a `String` (a URL) or a supported `Symbol` corresponding to a URL in the
-  # formula (e.g., `:stable`, `:homepage`, or `:head`).
+  # formula/cask (e.g., `:stable`, `:homepage`, `:head`, `:cask_url`, `:appcast`).
   # @param val [String, Symbol] URL to check for version information
   # @return [String, nil]
   def url(val = nil)
     @url = case val
     when nil
       return @url
+    when :cask_url
+      @formula_or_cask.url.to_s
+    when :appcast
+      @formula_or_cask.appcast.to_s
     when :head, :stable
-      @formula.send(val).url
+      @formula_or_cask.send(val).url
     when :homepage
-      @formula.homepage
+      @formula_or_cask.homepage
     when String
       val
     else
       raise TypeError, "Livecheck#url expects a String or valid Symbol"
     end
   end
 
+  # TODO: documentation
+  def version(val = nil)
```

Comment:
I think this should be reworked to have a more comprehensible value. Right now, if you do `some_cask.livecheck.version`, it returns something like `[":", :first]`, which is functional but not very valuable for debugging.

The two main options in my view are:

1. Store the symbol or string (not an array that's only useful internally).
2. Compute the version string within `version`, so we're always storing a string.

The first option is kind of similar to how `strategy` works, with the added wrinkle of potentially being a string. The second option is like how `url` works.

I'm beginning to think that the values in the `livecheck` DSL object should simply represent the values that are in the `livecheck` block itself. That is to say, `@version` should be the symbol or string found in the `livecheck` block. If `url` did the same thing, we wouldn't need to store the formula/cask object within the `livecheck` object.

This would address a shortcoming that I've found with the current `url` setup, where we may use `url :stable` in a formula's `livecheck` block but we can't tell that the value was the `:stable` symbol. Being able to know this is useful for the debug and verbose JSON output. It's possible to determine if the `stable` URL and `livecheck.url` match but we can't know that the `livecheck` block contained `url :stable`, which is a problem.

If we agree on this, I can take care of updating the `url` setup in a separate PR and this can be rebased on it, as I think that it's a bit outside the scope of this PR. The translation of a `url` symbol to the URL string from the formula would simply be done in livecheck itself rather than the DSL.

---

Besides that, it would be preferable to avoid duplicating the functionality from `Cask::DSL::Version` if we can help it. When it comes to computing the version string with respect to `before_comma`, `after_comma`, etc., is it possible to leverage the methods from there or does the way those methods are implemented prevent that?

---

Thoughts on all this?

Reply:
> I'm beginning to think that the values in the `livecheck` DSL object should simply represent the values that are in the `livecheck` block itself.

Yes, I think option 1 sounds better

> If `url` did the same thing, we wouldn't need to store the formula/cask object within the `livecheck` object.

Would it be useful to add a method to the `Livecheck` class that takes a formula/cask parameter and then computes the `url`/`version`?

> When it comes to computing the version string with respect to `before_comma`, `after_comma`, etc., is it possible to leverage the methods from there or are we prevented from doing that due to the way they're implemented?

It might be possible to use `Cask::DSL::Version` but I'll have to look into this